packages/field-base/src/input-mixin.js

```diff
@@ -126,6 +136,15 @@ export const InputMixin = dedupingMixin(
         }
       }
 
+      /**
+       * Observer to notify about the change of private property.
+       *
+       * @private
+       */
+      _hasInputValueChanged() {
+        this.dispatchEvent(new CustomEvent('has-input-value-changed'));
```

Comment:
I'm wondering if the event should contain the value.. this would allow to let the server parse the value and react differently depending on the value

Reply:
We were considering this option. 
Conclusion was that parsing logic for the value is already implemented for the component. This is for the cases when the value cannot be parsed, but flow counterpart should still be aware if there is some arbitrary content in the input field. (An example would be `qwerty` string in `date-picker`)

Reply:
This reminded me of another feature request, where `today` would be send to the server in this case and the server can parse it, like it was possible with V8. That why I wanted to mention it.

Reply:
You are right, that's a valid point, but that feature should be tackled separately (not in scope of this PR).
Related ticket: https://github.com/vaadin/flow-components/issues/2466